### DISCUSSION TITLE
Combat, spell cost reduction is not taken into account in the information window of the active spell on the creatures troop

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#4726] [Ideas]
In fheroes2, during combat, the spell cost reduction is not taken into account in the spell information window of the active spell on the creatures troop.

In fheroes2:

Crodo has the Evil Eye artifact that reduces the casting cost of Curse spells by half.
![Spell cost reduction 001](https://user-images.githubusercontent.com/43583994/143878714-201ef9ac-99e4-45a8-b178-00ab5affe369.png)

Myra has the Skullcap artifact that halves the casting cost of all mind influencing spells.
![Spell cost reduction 002](https://user-images.githubusercontent.com/43583994/143878728-aa77a547-a2a8-49a8-b59f-25e80b1cc201.png)

Myra has the Snake-Ring artifact that halves the casting cost of all Bless spells.
![Spell cost reduction 003](https://user-images.githubusercontent.com/43583994/143878737-3e0de0bd-b3e1-44ec-94f3-b425df0099e4.png)

However, this is not taken into account when displaying the cost of active spells in the spell information window during combat.

In fheroes2:

![Spell cost reduction 004](https://user-images.githubusercontent.com/43583994/143878834-7ce53883-4043-4f6f-92f8-9d79dfd4ce00.png)
![Spell cost reduction 005](https://user-images.githubusercontent.com/43583994/143878851-ef39db26-fbb3-4e2e-afac-ec9b30c6748a.png)

![Spell cost reduction 006](https://user-images.githubusercontent.com/43583994/143878858-8758165f-fb26-4888-91e1-10e02da43a2e.png)
![Spell cost reduction 007](https://user-images.githubusercontent.com/43583994/143878870-04c5f85b-26b8-4cfe-918c-f725b3b6d7fb.png)

![Spell cost reduction 008](https://user-images.githubusercontent.com/43583994/143878885-6838ab2b-a3aa-405d-ba4f-bedd9049f6d6.png)
![Spell cost reduction 009](https://user-images.githubusercontent.com/43583994/143878898-5c8a8637-f738-4c7c-ade3-58ee7700b022.png)

Since the player knows the cost of his spells and can deduct the cost of the spells cast by the opponent's hero by consulting the remaining spell points of the opponent's hero before and after casting the spell, I suggest to make it easier for him by displaying directly the modified costs of the active spells in the spell information window.

Related discussion: https://github.com/ihhub/fheroes2/discussions/4689.

**Comment by vincent-grosbois** [comment]
I think this could be transformed into a github issue
Also how about changing the "color" of the "[3]" to something like green/red depending if it is currently lowered or augmented ?

**Reply by LeHerosInconnu** [reply]
Hello @vincent-grosbois,

> I think this could be transformed into a github issue
> Also how about changing the "color" of the "[3]" to something like green/red depending if it is currently lowered or augmented ?

There is nothing that increases the cost of spells in Heroes 2 or in fheroes2... ...for now. :)
But yes, a color code can also be used to bring the player's attention to this, good idea. :)

Something like that:

For the hero whose troop is under the spell, the number of turns for which the spell will still be active, in this case "3", can be displayed in red to indicate to the player that the spell has a negative modification (for him) compared to its normal use.
Here, the spell costs only one spell point to be cast by the opposing hero.
![Spell cost reduction info red 001](https://user-images.githubusercontent.com/43583994/144061343-49c454e5-6992-4c46-a783-9a62a46ddd7c.png)

For the hero who cast the spell on the opposing troop, the number of turns for which the spell will still be active, in this case "3", can be displayed in green to indicate to the player that the spell has a positive modification compared to its normal use.
Here, the spell costs only one spell point to be cast by the hero.
![Spell cost reduction info green 001](https://user-images.githubusercontent.com/43583994/144061393-d3f7b913-2137-4193-a02c-4a7e499c19a2.png)

The spell cost number "[1]" is displayed in red in the troop's spell information window when it is the hero whose troop is under the spell that displays the spell information window.
![Spell cost reduction 1 red 005](https://user-images.githubusercontent.com/43583994/144061443-3fe027a4-0e01-4d94-b3cd-7ab773c3003e.png)

And the spell cost number "[1]" is displayed in green in the troop's spell information window when it is the hero who cast the spell on the opposing troop that displays the spell information window.
![Spell cost reduction 1 green 005](https://user-images.githubusercontent.com/43583994/144061548-7c9ae561-bd8d-4c5e-9ffa-77c7a5891ce6.png)

In the information bar at the bottom of the screen and in the logs, it can also be specified that the spell was cast for a reduced cost.
(No changes made in this image).
![Spell cost reduction info 004](https://user-images.githubusercontent.com/43583994/144061696-cddf781d-454d-4b05-ac8b-6f61569c365e.png)

Same procedure for the other spells, of course.

**Comment by LeHerosInconnu** [comment]
This is planned in my mod: https://github.com/ihhub/fheroes2/discussions/10130.
I close the discussion.
